assets/media-gallery.js

```diff
@@ -0,0 +1,101 @@
+if (!customElements.get('media-gallery')) {
+  customElements.define('media-gallery', class MediaGallery extends HTMLElement {
+    constructor() {
+      super();
+      this.elements = {
+        liveRegion: this.querySelector('[id^="GalleryStatus"]'),
+        viewer: this.querySelector('[id^="GalleryViewer"]'),
+        thumbnails: this.querySelector('[id^="GalleryThumbnails"]')
+      }
+      this.mql = window.matchMedia('(min-width: 750px)');
+      if (!this.elements.thumbnails) return;
+
+      this.elements.viewer.addEventListener('slideChanged', debounce(this.onSlideChanged.bind(this), 500));
+      this.elements.thumbnails.querySelectorAll('[data-target]').forEach((mediaToSwitch) => {
+        mediaToSwitch.querySelector('button').addEventListener('click', this.setActiveMedia.bind(this, mediaToSwitch.dataset.target, false));
+      });
+      if (this.dataset.desktopLayout !== 'stacked' && this.mql.matches) this.removeListSemantic();
+    }
+
+    onSlideChanged(event) {
+      const thumbnail = this.elements.thumbnails.querySelector(`[data-target="${ event.detail.currentElement.dataset.mediaId }"]`);
+      this.setActiveThumbnail(thumbnail);
+    }
+
+    setActiveMedia(mediaId, prepend) {
+      const activeMedia = this.elements.viewer.querySelector(`[data-media-id="${ mediaId }"]`);
+      this.elements.viewer.querySelectorAll('[data-media-id]').forEach((element) => {
+        element.classList.remove('is-active');
+      });
+      activeMedia.classList.add('is-active');
+
+      if (prepend) {
+        activeMedia.parentElement.prepend(activeMedia);
+        if (this.elements.thumbnails) {
+          const activeThumbnail = this.elements.thumbnails.querySelector(`[data-target="${ mediaId }"]`);
+          activeThumbnail.parentElement.prepend(activeThumbnail);
+        }
+        if (this.elements.viewer.slider) this.elements.viewer.resetPages();
+      }
+
+      this.preventStickyHeader();
+      window.setTimeout(() => {
+        if (this.elements.thumbnails) {
+          activeMedia.parentElement.scrollTo({ left: activeMedia.offsetLeft });
+        }
+        if (!this.elements.thumbnails || this.dataset.desktopLayout === 'stacked') {
+          activeMedia.scrollIntoView({behavior: 'smooth'});
+        }
```

Comment:
I kinda feel like it's nice to have it scroll back to the top of the media being opened. But I guess it's weird to keep focus on the thumbnail button and potentially have it outside of the viewport 🤔 

This can be a follow up conversation anyway. Not a blocker and quick fix if needed.
[video](https://screenshot.click/21-12-eg9gh-7dj93.mp4)

ccing @melissaperreault for future conversation on this.